fix: yarn classic  Command "tsc" not found

0618 opened the pull request:
<!--
Thank you for your Pull Request! Please describe the problem this PR fixes and a summary of the changes made.
Link to any relevant issues, code snippets, or other PRs.

For trivial changes, this template can be ignored in favor of a short description of the changes.
-->

## Problem

<!--
Describe the issue this PR is solving
-->

We removed installing `tsc` for yarn-classic in https://github.com/aws-amplify/amplify-backend/pull/1016 and the [test didn't fail](https://github.com/aws-amplify/amplify-backend/actions/runs/7906775269/job/21583388198), because the `yarn create amplify` was not using the correct beta version.

![image](https://github.com/aws-amplify/amplify-backend/assets/11983489/8f925ffe-bf62-4086-8f56-976aefad358d)

This PR is to fix yarn-classic and the test.

Note: we have to install `create-amplify` saperately because [yarn create fails when specifying a package version ](https://github.com/yarnpkg/yarn/issues/6587)



**Issue number, if available:**

## Changes

<!--
Summarize the changes introduced in this PR. This is a good place to call out critical or potentially problematic parts of the change.
-->

- install `tsc` for yarn-classic
- install `create-amplify` in local folder and use the binary directly

**Corresponding docs PR, if applicable:**

## Validation

<!--
Describe how changes in this PR have been validated. This may include added or updated unit, integration and/or E2E tests, test workflow runs, or manual verification. If manual verification is the only way changes in this PR have been validated, you will need to write some automated tests before this PR is ready to merge.

For changes to test infra, or non-functional changes, tests are not always required. Instead, you should call out _why_ you think tests are not required here.

If changes affect a GitHub workflow that is not included in the PR checks, include a link to a passing test run of the modified workflow.
--->

## Checklist

<!--
These items must be completed before a PR is ready to be merged.
Feel free to publish a draft PR before these items are complete.
-->

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [x] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

edwardfoyle commented on packages/integration-tests/src/package_manager_sanity_checks.test.ts:
```diff
@@ -81,13 +76,11 @@ void describe('getting started happy path', async () => {
-        tempDir
-      ).run();
+      await execa('yarn', ['add', 'create-amplify@beta'], { cwd: tempDir });
+      await execaCommand('./node_modules/.bin/create-amplify --yes --debug', {
```
could we run `create amplify@beta --yes` here instead of reaching into the bin folder?

0618 replied:
no, because 1) `create amplify` uses global installed package from `~/.config/yarn/global/`, 2) yarn-classic `create` doesn't accept tags 😂  https://github.com/yarnpkg/yarn/issues/6587

Amplifiyer commented:
Can you update the description with details on the following?
1. Why we removed installing the `tsc` from yarn classic in https://github.com/aws-amplify/amplify-backend/pull/1016?
2. Why do we have to add it back?

0618 commented:
> Can you update the description with details on the following?
> 
> 1. Why we removed installing the `tsc` from yarn classic in [fix(package-manager): pnpm tsc not found #1016](https://github.com/aws-amplify/amplify-backend/pull/1016)?
> 2. Why do we have to add it back?

It was removed because the test didn't fail after I removed them 😂  (I thought we could just remove)

Now I fixed the test and added them back.
